### PR TITLE
LPS-110450 Must use service.ranking for aggregation order

### PR DESCRIPTION
**Pull request by stian-sigvartsen**
Hi Preston. The ordering of aggregates is very important when overriding language keys. I have tweaked your suggestion to use ```service.ranking``` instead of ```aggregateId``` which is arbitrarily assigned based on deploy order.

I have included the new SaaS language keys module we are working on (and ready to merge actually). It is aggregating and overriding keys from ```saml-web``` which is already being extended by the aggregate registered by ```saml-addon-keep-alive-web``` . It's a complex scenario so hopefully will help with testing any solution.

The "fix" I am sending here does have a known shortcoming. Because the filter uses the ```service.ranking``` in the ```resource.bundle.aggregate``` directive instead of considering each filter in the aggregate separately.  By example, if you are rendering a language key from ```saml-web``` through the ServletContext and have the following...
```
liferay.resource.bundle;\
  bundle.symbolic.name=A;\
  resource.bundle.aggregate=(bundle.symbolic.name=A),(bundle.symbolic.name=B);\
  service.ranking=1;\
  servlet.context.name=saml-web
```
Then it would actually not consider keys in the following module C ...
```
liferay.resource.bundle;\
  bundle.symbolic..name=B;\
  resource.bundle.aggregate=(bundle.symbolic.name=C),(bundle.symbolic.name=B);\
  service.ranking=2
```
In other words you need to be aware of other aggregating modules when deciding on the correct service.ranking. You get a similar issue if two aggregates have the same ```service.ranking``` also.

The original code that get refactored took all this into account so probably the best solution would be to revert.

/cc @csierra

### CODE REVIEW

**Comment by Preston-Crary** [comment]
I think the same shortcoming existed before.  Going to combine both filters and send it back to you.

**Comment by Preston-Crary** [comment]
Please see https://github.com/stian-sigvartsen/liferay-portal/pull/239

**Comment by Preston-Crary** [comment]
728960b is handled by the aggregateId.
